### PR TITLE
Join declaration and assignment

### PR DESCRIPTION
**Pull request by hjmjohnson**
Declaration and assignment can be joined
    
Use IDE static checker to identify and fix variables that
can be initialized rather than declared and assigned.

This was a tedious semi-automated process.

Followup commits were changes that could be found
by using the joined declaration and assignment.

## PR Checklist
- [x] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [x] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)

### CODE REVIEW

File: Modules/Core/Transform/test/itkTransformTest.cxx [hunk 1]
```diff
@@ -190,7 +190,8 @@ class TransformTester
     transform->TransformPoint(pnt);
     std::cout << "TransformPoint()                              OK" << std::endl;
 
-    const InputVectorType vec{} {};
+    const InputVectorType vec{};
+    {};
```
[the remark this answers is not in the document]
Comment (hjmjohnson) [reply Modules/Core/Transform/test/itkTransformTest.cxx:194]: Resolved.

File: Modules/Core/Common/src/itkMetaDataDictionary.cxx [hunk 2]
```diff
@@ -44,10 +44,10 @@ void
 MetaDataDictionary::Print(std::ostream & os) const
 {
   os << "Dictionary use_count: " << m_Dictionary.use_count() << std::endl;
-  for (auto it = m_Dictionary->begin(); it != m_Dictionary->end(); ++it)
```
[the remark this answers is not in the document]
Comment (hjmjohnson) [reply Modules/Core/Common/src/itkMetaDataDictionary.cxx:47]: fixed.

**Review by dzenanz — APPROVED** [review]

File: Modules/Core/Transform/test/itkTransformTest.cxx (hunk 1 above)
[the remark this answers is not in the document]
Comment (hjmjohnson) [reply Modules/Core/Transform/test/itkTransformTest.cxx]: Fixed.

File: Modules/Core/Common/src/itkMetaDataDictionary.cxx (hunk 2 above)
[the remark this answers is not in the document]
Comment (hjmjohnson) [reply Modules/Core/Common/src/itkMetaDataDictionary.cxx:47]: Fixed.

**Review by dzenanz — APPROVED** [review]

**Comment by jhlegarreta** [comment]
Thanks for all the work put lately to make the style in ITK more streamlined with recommendations and more consistent @hjmjohnson :100: and sorry for not being able to contribute with PRs lately.

The dashboard has started complaining about changes related to this PR, e.g.
https://open.cdash.org/viewBuildError.php?buildid=10096096

**Comment by hjmjohnson** [comment]
@jhlegarreta  Weird. I did not see that on my builds.  ... Fixed by initializing in next PR related to const and constexpr.

**Pull request merged**